code/game/gamemodes/distress.dm

```diff
@@ -0,0 +1,695 @@
+/datum/game_mode/distress
+	name = "Distress Signal"
+	config_tag = "Distress Signal"
+	required_players = 2
+	flags_round_type = MODE_INFESTATION|MODE_FOG_ACTIVATED
+	flags_landmarks = MODE_LANDMARK_SPAWN_XENO_TUNNELS|MODE_LANDMARK_SPAWN_MAP_ITEM
+
+	var/list/survivors = list()
+
+	var/xeno_required_num 	  = 1
+	var/xeno_starting_num 	  = 0
+	var/surv_starting_num 	  = 0
+	var/merc_starting_num 	  = 0
+	var/marine_starting_num   = 0
+
+	var/bioscan_current_interval = 45 MINUTES
+	var/bioscan_ongoing_interval = 20 MINUTES
+
+	var/list/xenomorphs = list()
+	var/datum/mind/queen
+	var/stored_larva 		= 0
+	var/latejoin_tally		= 0
+	var/latejoin_larva_drop = 0
+	var/queen_death_countdown = 0
+
+
+/datum/game_mode/distress/can_start()
+	. = ..()
+	initialize_scales()
+	var/found_queen = initialize_queen()
+	var/found_xenos = initialize_xenomorphs()
+	if(!found_queen && !found_xenos)
+		return FALSE
+	initialize_survivor()
+	latejoin_larva_drop = CONFIG_GET(number/latejoin_larva_required_num)
+	return TRUE
+
+
+/datum/game_mode/distress/announce()
+	to_chat(world, "<span class='round_header'>The current map is - [SSmapping.config.map_name]!</span>")
+
+
+/datum/game_mode/distress/post_setup()
+	. = ..()
+
+	for(var/i in xenomorphs)
+		var/datum/mind/M = i
+		if(M == queen)
+			transform_queen(M)
+		else
+			transform_xeno(M)
+
+	for(var/i in survivors)
+		var/datum/mind/M = i
+		transform_survivor(M)
+
+	scale_gear()
+
+	addtimer(CALLBACK(SSticker.mode, .proc/map_announce), 5 SECONDS)
+
+
+
+/datum/game_mode/distress/proc/map_announce()
+	switch(SSmapping.config.map_name)
+		if(MAP_LV_624)
+			command_announcement.Announce("A faint distress signal has been picked up by our scanners, which have tracked the source to a third generation colony, known as LV-624. Through use of bluespace drive tech, the [CONFIG_GET(string/ship_name)] has jumped within range of the colony. TGMC, gear up and get ready to respond!", "[CONFIG_GET(string/ship_name)]")
```

Comment:
this should probably be moved into the map configs.

Reply:
It's gamemode specific though

Reply:
It's map specific like you could store this text in the map json and replace the switch with just a reference to map config. Ideally we should be able to just add maps via config without having to change code like this.